src/services/firebase/firestore.ts

```diff
@@ -113,26 +115,30 @@ export class FirestoreServices {
       conversationData.image = image;
     }
     /** Create the conversation to the user who create the chat */
-    const conversationRef = await firestore()
-      .collection<Partial<ConversationProps>>(
-        `${FireStoreCollection.users}/${this.userId}/${FireStoreCollection.conversations}`
-      )
-      .add(conversationData);
+    const conversationRef = firestore().collection<Partial<ConversationProps>>(
+      `${FireStoreCollection.users}/${this.userId}/${FireStoreCollection.conversations}`
+    );
+
+    if (conversationId) {
+      await conversationRef.doc(conversationId).set(conversationData);
+    } else {
+      newConversationId = (await conversationRef.add(conversationData)).id;
+    }
     /** Add the conversation to the user who is conversation's member */
     await Promise.all([
       memberIds.map((memberId) => {
         return firestore()
           .collection(
             `${FireStoreCollection.users}/${memberId}/${FireStoreCollection.conversations}`
           )
-          .doc(conversationRef.id)
+          .doc(newConversationId)
```

Comment:
We can use `conversationId`

Reply:
In case we don't pass `conversationId`, I have to assign newly created document to this variable